sinatra-contrib/lib/sinatra/capture.rb

```diff
@@ -106,12 +105,12 @@ def capture(*args, &block)
         dummy      = DUMMIES.fetch(current_engine)
         options    = { :layout => false, :locals => {:args => args, :block => block }}
 
-        buffer.try :clear
+        buffer.clear if buffer
```

Comment:
I think Object#try checks to see if the receiver responds to the given method, not merely if the receiver is not falsey. How about:

```suggestion
        buffer.clear if buffer.respond_to?(:clear)
```

Reply:
You are correct. I probably assumed to much based on an earlier `if buffer` check in the file.